source/common/stream_info/stream_info_impl.h

```diff
@@ -343,6 +349,7 @@ struct StreamInfoImpl : public StreamInfo {
   std::string upstream_transport_failure_reason_;
   absl::optional<Upstream::ClusterInfoConstSharedPtr> upstream_cluster_info_;
   absl::optional<uint64_t> connection_id_;
+  std::string connection_fingerprint_;
```

Comment:
Should this use `optional` like the connection ID does, rather than returning an empty string if its's not set?